Dockerfile

```diff
@@ -1,21 +1,22 @@
 #TODO pin all versions to sha digest
 #TODO add .dockerignore
-FROM node:20.11-slim as nodebuilder
+FROM node@sha256:ecc9a2581f8588014a49a523a9ed146d27963f6d988d11bd16bbdcb3598f5f98 as nodebuilder
 WORKDIR /opt/app
 COPY package-lock.json ./
 COPY package.json ./
 RUN npm install
 
-FROM ruby:3.2 as rubybuilder
+FROM ruby@sha256:ddc5729409d1d3222e74a5edb62d142ebd5fa47e9a98fc2905d66056eec6ae3b as rubybuilder
 RUN apt update -y && apt -y install rsync nano
 RUN cp /usr/bin/nano /usr/local/bin/
+
 WORKDIR /opt/app
 COPY Gemfile Gemfile.lock ./
 COPY --from=nodebuilder /usr/local/bin /usr/local/nodebin
 RUN rsync -a /usr/local/nodebin /usr/local/bin
 RUN bundle install
 
-FROM ruby:3.2-slim
+FROM ruby@sha256:04da59d84a16b6db4a6663a6940a5142d79a50d8727acd00f10c3701cdeb46b0
```

Comment:
```
 docker pull ruby:3.2-slim
3.2-slim: Pulling from library/ruby
Digest: sha256:c09c3ea897b403f8fdd43957c3d3777aa2c0b991f4f4194b26cbfd9eb62fca33
Status: Image is up to date for ruby:3.2-slim
docker.io/library/ruby:3.2-slim
```

Reply:
I've checked all 3.2-slims (3.2.3-slim, 3.2.2-slim, 3.2.1-slim, 3.2.0-slim) - none of them are the sha in the pull request.  - what is this shadigest from?

Reply:
@tabroughton I used this command `docker inspect --format='{{index .RepoDigests 0}}' ruby:3.2-slim`. This is the output I have for `ruby:3.2-slim`. `ruby@sha256:04da59d84a16b6db4a6663a6940a5142d79a50d8727acd00f10c3701cdeb46b0`.
This is the [stackoverflow ](https://stackoverflow.com/questions/32046334/where-can-i-find-the-sha256-code-of-a-docker-image)link to the command.

Reply:
Tom, I think the sha Hauwa has used _is_ from a ruby:3.2-slim image that was [built on 7 Feb 2024](https://github.com/docker-library/repo-info/commit/d60cb7e9bf1695258bc9aa19b9c0abe3eb263198#diff-69d0dc86abb92e89b92ccd88923c443fd536d6ff91c0650f2890f8d833f289cfR4) and [updated on 13 Feb 2024](https://github.com/docker-library/repo-info/commit/f2d392bf421b75e1635481c02af5866b2d7418d4#diff-69d0dc86abb92e89b92ccd88923c443fd536d6ff91c0650f2890f8d833f289cfL4). This can happen when eg. a  package in the underlying OS is updated. You can see from the records in [repo-info](https://github.com/docker-library/repo-info) that the sha was current for the package on [7 Feb](https://github.com/docker-library/repo-info/blob/d60cb7e9bf1695258bc9aa19b9c0abe3eb263198/repos/ruby/remote/3.2-slim.md)